### PR TITLE
Add IONOS API support

### LINKED ISSUE #3284: Question: Ionos DNS API
Hello everyone,

I would like to ask if there are any plans to support the ionos (earlier 1und1) dns api?
https://beta.developer.hosting.ionos.de/docs/dns

Thanks in advance.

### PR DESCRIPTION
**Pull request by lbrocke**
This PR adds support for the [IONOS DNS API](https://developer.hosting.ionos.de/docs). ACME v2 wildcard certificates are supported, [all tests are passing on my fork](https://github.com/lbrocke/acme.sh/actions/runs/499459718).

Please let me know if any changes are desired. Otherwise I'm happy to update the docs and create an issue for bug reports.

Closes #3284

### CODE REVIEW

File: dnsapi/dns_ionos.sh [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+  if ! _get_root "$fulldomain"; then
+    _err "Cannot find this domain in your IONOS account."
+    return 1
+  fi
+}
+
+_get_root() {
+  domain=$1
+  i=2
```
Comment (Neilpang) [inline dnsapi/dns_ionos.sh:95]: `i`  should start from `1`

**Comment by Neilpang** [comment]
>> Otherwise I'm happy to update the docs and create an issue for bug reports.

Please go ahead.

File: dnsapi/dns_ionos.sh (hunk 1 above)
Comment (lbrocke) [reply dnsapi/dns_ionos.sh]: Fixed

**Comment by lbrocke** [comment]
> > > Otherwise I'm happy to update the docs and create an issue for bug reports.
> 
> Please go ahead.

I updated the documentation (see [here](https://github.com/acmesh-official/acme.sh/wiki/dnsapi#129-using-the-ionos-domain-api)) and created an issue for bug reports (#3379). The desired change is also now implemented.

**Pull request merged**